### PR TITLE
Cleanup error handling in format_strings.cpp

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 1dd6f70bd890377bab9cd847fc268a286164b52f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84841239

**Review by peterschrammel — APPROVED** [review]

File: src/goto-programs/format_strings.cpp [hunk 1]
```diff
@@ -175,7 +178,8 @@ void parse_conversion_specifier(
     }
 
     default:
-      throw std::string("unsupported format conversion specifier: `")+*it+"'";
+      throw deserialization_exceptiont(
+        std::string("unsupported format conversion specifier: `") + *it + "'");
   }
```
Comment (kroening) [inline src/goto-programs/format_strings.cpp]: This is entirely unrelated to deserialisation.

**Comment by kroening** [comment]
The system_error_exception introduced here appears to be unrelated to anything in format_strings.cpp.

**Comment by chrisr-diffblue** [comment]
Rebased to pickup newer exception classes and now using one of the new exception classes as a more appropriate exception type.

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: fead175ce56cceee6fd73fa47f2be08451ce6276).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85445548
